gapis/memory/types.go

```diff
@@ -18,21 +18,45 @@ import "reflect"
 
 var (
 	tyPointer = reflect.TypeOf((*Pointer)(nil)).Elem()
-	tyChar    = reflect.TypeOf(Char(0))
-	tyInt     = reflect.TypeOf(Int(0))
-	tyUint    = reflect.TypeOf(Uint(0))
-	tySize    = reflect.TypeOf(Size(0))
+	tyChar    = reflect.TypeOf((*CharTy)(nil)).Elem()
```

Comment:
Sorry, I misunderstood. I will rename them, but won't add the original type checking back.